### PR TITLE
Exception Review: JwtSecurityTokenHandler.ValidateToken, OIDCProtocol…

### PR DESCRIPTION
**Pull request by brentschmaltz**
…Validator

Fixed some typo's
Added clarifying exceptions messages
AtHash and CHash both use string.IsNullOrWhiteSpace()

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/375

### CODE REVIEW

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/LogMessages.cs [hunk 1]
```diff
@@ -92,6 +92,7 @@ internal static class LogMessages
         internal const string IDX10346 = "IDX10346: OpenIdConnectProtocolValidationContext.ValidatedIdToken does not contain a 'sub' claim, cannot validate.";
         internal const string IDX10347 = "IDX10347: Validating the 'c_hash' failed, see inner exception.";
         internal const string IDX10348 = "IDX10348: Validating the 'at_hash' failed, see inner exception.";
+        internal const string IDX10349 = "IDX10349: The id_token: '{0}' is not valid. Delegate threw exception, see inner exception for more details.";
```
Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/LogMessages.cs]: Please update IDX10313 if its not used.

**Comment by brentschmaltz** [comment]
@tugupta updated, reused IDX10313

File: src/Microsoft.IdentityModel.Tokens/LogMessages.cs [hunk 2]
```diff
@@ -46,26 +46,29 @@ internal static class LogMessages
...
         internal const string IDX10236 = "IDX10236: Issuer Validated.Issuer: '{0}'";
-        internal const string IDX10237 = "IDX10237: ValidateIssuerSigningKey property on ValidationParamaters is set to false. Exiting without validating the issuer signing key.";
-        internal const string IDX10238 = "IDX10238: ValidateLifetime property on ValidationParamaters is set to false. Exiting without validating the lifetime.";
+        internal const string IDX10237 = "IDX10237: ValidateIssuerSigningKey property on ValidationParameters is set to false. Exiting without validating the issuer signing key.";
+        internal const string IDX10238 = "IDX10238: ValidateLifetime property on ValidationParameters is set to false. Exiting without validating the lifetime.";
         internal const string IDX10239 = "IDX10239: Lifetime of the token is valid.";
         internal const string IDX10240 = "IDX10240: No token replay is detected.";
+        internal const string IDX10241 = "IDX10241: Unable to validate IssuerSecurityKey. validationParameters.IssuerSigningKeyValidator is null. By default ValidateIssuerSigningKey is false, if set to true, theIssuerSigningKeyValidator must be set.";
```
Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Tokens/LogMessages.cs]: nit: theIssuerSigningKeyValidator => the IssuerSigningKeyValidator

**Comment by tushargupta51** [comment]
:shipit:

**Comment by brentschmaltz** [comment]
@tushargupta51 thanks, Ill fix the typo.

**Pull request merged**